src/client/gameannouncer.py

```diff
@@ -1,61 +1,60 @@
+        self._friends_event_tracker.friendEvent.connect(self._friend_event)
 
         self.announce_games = True
         self.announce_replays = True
-        self._delayed_host_list = []
-
-    def _is_friend_host(self, game):
-        return (game.host_player is not None
-                and self._me.isFriend(game.host_player.id))
-
-    def _announce_hosting(self, game):
-        if not self._is_friend_host(game) or not self.announce_games:
-            return
-        announce_delay = QTimer()
-        announce_delay.setSingleShot(True)
-        announce_delay.setInterval(self.ANNOUNCE_DELAY_SECS * 1000)
-        announce_delay.timeout.connect(self._delayed_announce_hosting)
-        announce_delay.start()
-        self._delayed_host_list.append((announce_delay, game))
+        self._delayed_event_list = []
+        self.delay_friend_events = True
 
-    def _delayed_announce_hosting(self):
-        timer, game = self._delayed_host_list.pop(0)
+    def _friend_event(self, player, event):
+        if self.delay_friend_events:
+            self._delayed_event_list.append((player, event))
+        else:
+            self._friend_announce(player, event)
 
-        if (not self._is_friend_host(game) or
-           not self.announce_games or
-           game.state != GameState.OPEN):
-            return
-        self._announce(game, "hosting")
+    def delayed_friend_events(self):
+        self.delay_friend_events = False
+        while len(self._delayed_event_list) > 0:
+            player, event = self._delayed_event_list.pop(0)
+            self._friend_announce(player, event)
 
-    def _announce_replay(self, game):
-        if not self._is_friend_host(game) or not self.announce_replays:
+    def _friend_announce(self, player, event):
+        if player.currentGame is None:
             return
-        self._announce(game, "playing live")
-
-    def _announce(self, game, activity):
-        url = game.url(game.host_player.id).toString()
-        url_color = self._colors.getColor("url")
-        mapname = maps.getDisplayName(game.mapname)
-        fmt = 'is {} {}<a style="color:{}" href="{}">{}</a> (on {})'
+        game = player.currentGame
+        player_info = ""
+        if game.state == GameState.OPEN:
+            if not self.announce_games:  # Menu Option Chat
+                return
+            if game.featured_mod == "ladder1v1":
+                activity = "started"
+            elif player.login == game.host:
+                activity = "is <font color='GoldenRod'>hosting</font>"
+            else:
+                activity = "joined"
+            player_info = " [{}/{}]".format(game.num_players, game.max_players)
+        elif game.state == GameState.PLAYING:
+            if not self.announce_replays:  # Menu Option Chat
+                return
+            activity = "is playing live"
+        else:  # GameSate.CLOSED
+            activity = "<font color='Red'>has left the building</font>"
```

Comment:
This *will* confuse people. Do we even need an event for a friend leaving a game?

Reply:
Well, that is the 'shouldn't happen' case, I like to have those cases signaled if it happens (makes a bigger splash than a log entry).
But I can live without it, cause I noticed you prefer coding without humor.

Reply:
Humour is nice if it doesn't seep into user experience. You're not the one who will have to deal with confused users on the tech support forum.

Reply:
But that's the way we'll probably hear that there's something unintended/wrong going on.
(and it's an Elvis reference, so hopefully someone will know it's not really faf context.)
If we skip the case, we'll never know, that might be fine too. If you want it gone, say so.

Reply:
It's not very unusual for it to happen, since between the event firing and the queue being freed the game could've ended. Either way it's still better to log it rather than bother the user with something a dev wants to know. Since it will happen occasionally anyway, I'd rather see it gone.

Reply:
the game ended case is handled on top. this is different, maybe you look at the updated code... very eventfull.